native/src/modules/app/containers/RedirectContainer.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+
+import React, { useEffect } from 'react'
+import type { NavigationPropType, RoutePropType } from '../constants/NavigationTypes'
+import type { RedirectRouteType } from 'api-client'
+import { useDispatch } from 'react-redux'
+import { useTranslation } from 'react-i18next'
+import LayoutContainer from '../../layout/containers/LayoutContainer'
+import navigateToDeepLink from '../../navigation/navigateToDeepLink'
+
+const TIMEOUT = 50
+
+type PropsType = {|
+  route: RoutePropType<RedirectRouteType>,
+  navigation: NavigationPropType<RedirectRouteType>
+|}
+
+const RedirectContainer = ({ route, navigation }: PropsType) => {
+  const dispatch = useDispatch()
+  const { i18n } = useTranslation()
+  const { language } = i18n
+  const { url } = route.params
+
+  useEffect(() => {
+    // If actions are dispatched/navigate is called to early it fails and nothing happens
+    // Therefore we wait for a short time period
+    // https://github.com/react-navigation/react-navigation/issues/8537
+    const timeout = setTimeout(() => {
+      navigateToDeepLink(dispatch, navigation, url, language)
+    }, TIMEOUT)
```

Comment:
Is it possible to do this in a loop and check whether navigation is ready?

A timeout with 50ms will certainly fail on older devices.